src/ConfigFetcher.php

```diff
@@ -116,7 +176,12 @@ public function fetch($etag)
                     $etag = $response->getHeader(self::ETAG_HEADER)[0];
                 }
 
-                return new FetchResponse(FetchResponse::FETCHED, $etag, $body);
+                $url = "";
+                if (isset($body[Config::PREFERENCES]) && isset($body[Config::PREFERENCES][Preferences::BASE_URL])) {
+                    $url = $body[Config::PREFERENCES][Preferences::BASE_URL];
+                }
+
+                return new FetchResponse(FetchResponse::FETCHED, $etag, $body, $url);
```

Comment:
I don't see where that $url is used in executeFetch(). What's the purpose of returning it?